### PR TITLE
doc/md: add tutorial on how to generate mutation input types

### PR DESCRIPTION
**Pull request by a8m**
Fixes https://github.com/ent/ent/issues/1429

### CODE REVIEW

File: doc/md/tutorial-todo-gql-mutation-input.md [hunk 1]
```diff
@@ -0,0 +1,422 @@
+---
+id: tutorial-todo-gql-mutation-input
+title: Mutation Inputs
+sidebar_label: Mutation Inputs
+---
+
+In this section, we continue the [GraphQL example](tutorial-todo-gql.md) by explaining how to extend the Ent code
+generator using Go templates and generate [input type](https://graphql.org/graphql-js/mutations-and-input-types/)
+objects for our GraphQL mutations that can be applied directly on Ent mutations.
```
Comment (rotemtam) [inline doc/md/tutorial-todo-gql-mutation-input.md:9]:
```suggestion
In this section, we continue the [GraphQL example](tutorial-todo-gql.md) by explaining how to extend the Ent code
generator using Go templates to generate [input type](https://graphql.org/graphql-js/mutations-and-input-types/)
objects for our GraphQL mutations that can be applied directly on Ent mutations.
```

File: doc/md/tutorial-todo-gql-mutation-input.md [hunk 2]
```diff
@@ -0,0 +1,422 @@
...
+cd ent-graphql-example 
+go run ./cmd/todo/
+```
+
+## Go Templates
+
+The Ent framework accepts external templates that can extend or override the default generated functionality of its 
+code generator. In the template below, we generate 2 input types (`CreateTodoInput` and `UpdateTodoInput`) for the
+GraphQL mutations, and add additional methods on the different builders to accept these objects as an input type.
+
```
Comment (rotemtam) [inline doc/md/tutorial-todo-gql-mutation-input.md:28]: i think its a bit hard to follow template code if you don't know what we're generating. Maybe add the final result (you can comment out most of if for brevity, but help the reader understand what's going to happen.)

File: doc/md/tutorial-todo-gql-mutation-input.md [hunk 3]
```diff
@@ -0,0 +1,422 @@
...
+        {{- end }}
+    }
+
+    {{/* Additional methods go here.  */}}
+{{ end }}
+```
+
+The full version of this template exists in the [repository](https://github.com/a8m/ent-graphql-example/blob/master/ent/template/mutation_input.tmpl).
+
+Now, we tell the Ent code generator to execute this template by passing it as an argument in the `ent/entc.go` file:
```
Comment (rotemtam) [inline doc/md/tutorial-todo-gql-mutation-input.md:75]: I think it will be useful to add an "info box" explaining a bit about the different "magic" (ie available in the template FuncMap) functions that are available. Give an example or two and link to where I can find the whole thing.

File: doc/md/tutorial-todo-gql-mutation-input.md (hunk 2 above)
Comment (rotemtam) [reply doc/md/tutorial-todo-gql-mutation-input.md:28]: In addition, I'd consider adding an example of some "hello world" code generation just to explain how the mechanism works (how entc / go templates / the template directory interact) before going into the actual GQL gen. WDYT?

Comment (a8m) [reply doc/md/tutorial-todo-gql-mutation-input.md:28]: > d to follow template code if you don't know what we're generating. Maybe add the final result (you can comment out most of if for brevity, but help the reader understand what's going to happen.)

It's there. See line 96. 

> In addition, I'd consider adding an example of some "hello world" code generation just to explain how the mechanism works (how entc / go templates / the template directory interact) before going into the actual GQL gen. WDYT?

mm.. maybe point them to the template-doc on our website instead?

**Comment by a8m** [comment]
Thanks for the review @rotemtam.
